Duplicati/Library/Utility/SubStream.cs

```diff
@@ -0,0 +1,118 @@
+﻿#region Disclaimer / License
+// Copyright (C) 2015, The Duplicati Team
+// http://www.duplicati.com, info@duplicati.com
+// 
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// 
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+// 
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace Duplicati.Library.Utility
+{
+    /// <summary>
+    /// This class wraps a Stream but only allows reading a limited number of bytes from the underlying stream.
+    /// This can be used to create split a source stream into multiple smaller buffers that can be used as independent streams.
+    /// </summary>
+    public class SubStream : Stream

```

Comment:
On the surface, this appears to have a similar purpose to [ReadLimitLengthStream](https://github.com/duplicati/duplicati/blob/v2.0.5.114-2.0.5.114_canary_2021-03-10/Duplicati/Library/Utility/ReadLimitLengthStream.cs).  The implementation is different, probably due to the different use cases.  If these indeed are meant to provide similar functionality, do you see any way to consolidate the two?

Reply:
I guess I should have looked harder for something that already did this - I didn't see that class. After a quick look, I think I can add my new features to that class and use it instead.

Reply:
Looks like my change is causing some tests to fail. I'll have to look more into it.

Reply:
Tests are fixed and some new tests are added to validate the new behavior on ReadLimitLengthStream.